tests/test_builders/test_build_warnings.py

```diff
@@ -17,15 +19,38 @@
 
     from sphinx.testing.util import SphinxTestApp
 
-ENV_WARNINGS = """\
-{root}/autodoc_fodder.py:docstring of autodoc_fodder.MarkupError:\\d+: \
+if version_info >= (11, 0, 0):
+    # this code can be removed when we prepare Sphinx 11 for release
+    msg = 'From Sphinx 11, duplicate labels should always warn by default'
+    raise RemovedInSphinx11Warning(msg)
+else:
+    DUPLICATE_LABEL_WARNINGS = version_info >= (10, 0, 0)
```

Comment:
This isn't the right approach: we shouldn't raise warnings as exceptions, nor does it make sense to raise a module-level warning in a test file.

Reply:
It doesn't make sense, but it would only occur when we begin developing Sphinx 11 -- the major release after the info-level messages are upgraded to warnings.  At that point, the test condition could be removed.

What's a better way to annotate test code that can be cleaned up after a certain version?